### PR TITLE
Upgrade torch version & remove rikai dataloader

### PR DESCRIPTION
**Pull request by Renkai**
See the details in issue https://github.com/eto-ai/rikai/issues/342

### CODE REVIEW

File: python/tests/torch/test_data.py [hunk 1]
```diff
@@ -19,103 +19,12 @@
...
-                "array": array,
-                "image": Image.from_array(image_data, image_uri),
-            }
-        )
-        expected.append({"id": i, "array": array, "image": image_data})
-    df = spark.createDataFrame(data)
-
-    df.write.mode("overwrite").format("rikai").save(str(dataset_dir))
-
-    loader = DataLoader(dataset_dir, batch_size=8)
```
[the remark this answers is not in the document]
Comment (Renkai) [reply python/tests/torch/test_data.py:55]: https://github.com/eto-ai/rikai/blob/88541f3140352161f9647c7f0caa5abfaac65442/python/tests/torch/test_data.py#L124
If you mean test `rikai.torch.Dataset` with `torch.utils.data.DataLoader`, I think it already exists in the current master branch. Do you mean more kinds of test? I don't know what different kinds of test can we add.

[the remark this answers is not in the document]
Comment (eddyxu) [reply python/tests/torch/test_data.py:55]: Ah, you are right.  

The test coverage is good enough for now. Will approve soon

**Review by eddyxu — APPROVED** [review]
💯 nice work! @Renkai

**Comment by da-tubi** [comment]
Rikai 0.0.12 can only work with Databricks Runtime 8.3.x (with Pytorch 1.8.x installed)

I think this PR can be merged along with the Spark 3.2.x pr.

**Pull request merged**